src/components/PinInputField/PinInputField.tsx

```diff
@@ -125,8 +125,12 @@ export const PinInputField = React.forwardRef(
 
             if (eventValue.length > 1 && i < length - 1) {
                 if (validate(eventValue, type)) {
-                    const nextValue = eventValue.split("").filter((_, index) => index < length);
-                    setValues(nextValue);
+                    const nextValue = eventValue.split("").filter((_, index) => index > 0 && i + index - 1 < length);
+                    setValues((values) =>
+                        values.map((v, index) =>
+                            index >= i && index < i + nextValue.length ? nextValue[index - i] : v
```

Comment:
can we update naming from `i` & `index` here? It's hard to read and can be confusing with both of these being used.

Reply:
Have renamed the InputElement iterator to `i` and the `nextValue` iterators to `j`. I'm not super sure if these are an improvement though. [Open to naming suggestions if they aren't : P](https://martinfowler.com/bliki/TwoHardThings.html)

Reply:
How about renaming the `i` param to `inputIndex` or `inputFieldIndex` for this function? And keeping other iteration indexes as `i` or `index` will be fine I guess.